src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs

```diff
@@ -220,118 +220,58 @@ public override IEnumerable<FileSystemInfo> EnumerateFileSystemInfos(string full
             }
         }
 
-        // Returns 0 on success, otherwise a Win32 error code.  Note that
-        // classes should use -1 as the uninitialized state for dataInitialized.
-        [System.Security.SecurityCritical]  // auto-generated
-        internal static int FillAttributeInfo(string path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool tryagain, bool returnErrorOnNotFound)
+        /// <summary>
+        /// Returns 0 on success, otherwise a Win32 error code.  Note that
+        /// classes should use -1 as the uninitialized state for dataInitialized.
+        /// </summary>
+        /// <param name="returnErrorOnNotFound">Return the error code for not found errors?</param>
+        [System.Security.SecurityCritical]
+        internal static int FillAttributeInfo(string path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool returnErrorOnNotFound)
         {
-            int errorCode = 0;
-            if (tryagain) // someone has a handle to the file open, or other error
+            int errorCode = Interop.Errors.ERROR_SUCCESS;
+
+            // Neither GetFileAttributes or FindFirstFile like trailing separators
+            path = path.TrimEnd(PathHelpers.DirectorySeparatorChars);
+
+            using (new DisableMediaInsertionPrompt())
```

Comment:
This is one of the highest hit APIs, incidentally- so the delegate perf thing worries me more than I would normally